### PR TITLE
feat: added v2 service as submodule

### PR DESCRIPTION
**Pull request by prabhu34**
- added `cloud_run_v2_service` submodule
- added example for v2 service submodule
- TODO: Test all features and add unit tests
- added environment variables as map instead of list in earlier module versions. Addresses #197
- addresses: #183 and #198 

@bharathkkb @q2w

### CODE REVIEW

File: modules/cloud-run-v2-service/main.tf
Comment (prabhu34) [reply modules/cloud-run-v2-service/main.tf:41]: There are scaling at template as well as service level. Wouldn't it be confusing to have that as a top level primitive? Thoughts?

File: modules/cloud-run-v2-service/variables.tf
Comment (prabhu34) [reply modules/cloud-run-v2-service/variables.tf:294]: Only one port is supported. So making it map would allow to add more ports. Hence leave it as object.

File: modules/cloud-run-v2-service/variables.tf
Comment (prabhu34) [reply modules/cloud-run-v2-service/variables.tf:276]: Yes, agreed. Made it map(object) with secret and version as objects.

File: modules/cloud-run-v2-service/versions.tf
Comment (prabhu34) [reply modules/cloud-run-v2-service/versions.tf:32]: Done

File: modules/cloud-run-v2-service/README.md
Comment (prabhu34) [reply modules/cloud-run-v2-service/README.md:19]: Done

File: modules/cloud-run-v2-service/main.tf
Comment (prabhu34) [reply modules/cloud-run-v2-service/main.tf:62]: Yes. I have encapsulated the variable into one which allows multiple container blocks.

File: modules/v2/variables.tf
Comment (q2w) [inline modules/v2/variables.tf:133]: Should this be list(object())? I can see this is being used in a dynamic block with intention of creating multi `vpc_access`.

File: modules/v2/main.tf
Comment (q2w) [inline modules/v2/main.tf:38]: Why are we keeping this as dynamic block? Are we expecting multiple scaling settings! Which one would take precedence?

File: modules/v2/README.md
Comment (q2w) [inline modules/v2/README.md:1]: Should a metadata.yaml file be created for this new module? I don't see metadata.yaml available for this module. But we do have this file for modules present in https://github.com/terraform-google-modules

**Comment by prabhu34** [comment]
@q2w Feel free to make changes and amend!

File: modules/v2/variables.tf
Comment (prabhu34) [reply modules/v2/variables.tf:133]: No. This is a single block.

File: modules/v2/main.tf
Comment (prabhu34) [reply modules/v2/main.tf:38]: Again, these are not multiple blocks. It's a single object similar to `vpc_access` and others.
There is `scaling` for each revision and overall service. Scaling for overall service splits the instances across revisions.

File: modules/v2/README.md
Comment (prabhu34) [reply modules/v2/README.md:1]: I am not very sure of a metadata.yaml files usage. @bharathkkb Should I generate them?

**Comment by prabhu34** [comment]
@bharathkkb For your second review with changes.

File: modules/v2/main.tf
Comment (bharathkkb) [inline modules/v2/main.tf:65]: Could we use key rather than using a separate container_name?

File: modules/v2/main.tf
Comment (bharathkkb) [inline modules/v2/main.tf:276]: Do we need to ignore chnage any annotations added by the service like
https://github.com/GoogleCloudPlatform/terraform-google-cloud-run/blob/b59b700a7a5d3c15847036792db88df7eb9503db/main.tf#L192

**Review by bharathkkb — APPROVED** [review]
Sorry for the delay! A few minor comments otherwise lgtm

File: modules/v2/README.md
Comment (bharathkkb) [reply modules/v2/README.md:1]: @prabhu34 you can set this in makefile to autogenerate https://github.com/terraform-google-modules/terraform-google-folders/blob/696570e8cf5d996336b586c9293ae6fea3edf98c/Makefile#L79

go/blueprints-metadata has more instructions but feel free to do this in a follow up to not block this PR

File: modules/v2/main.tf
Comment (prabhu34) [reply modules/v2/main.tf:276]: I guess these are not needed as they will be part of [effective_annotations](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloud_run_v2_service#effective_annotations) and we can add them later if really needed.

File: modules/v2/main.tf
Comment (prabhu34) [reply modules/v2/main.tf:65]: I tried that earlier, but having a container name is optional input and container image will be a bigger key name. Let me know your thoughts!

File: modules/v2/README.md
Comment (prabhu34) [reply modules/v2/README.md:1]: I have created the metadata now. Seems like the Makefile was outdated and I had added the env to the generate_docs.

File: modules/v2/variables.tf
Comment (q2w) [inline modules/v2/variables.tf:234]: Can we make this as list(object) as we are doing for volumes input variable? @prabhu34

Comment (prabhu34) [reply modules/v2/variables.tf]: Done

File: modules/v2/main.tf
Comment (prabhu34) [reply modules/v2/main.tf:65]: Made it list(objects) instead.

**Review by q2w — APPROVED** [review]

**Pull request merged**
